### PR TITLE
fix: cannot resolve paths with hash

### PR DESCRIPTION
**Pull request by picheli20**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/22979

## What is the new behavior?
It's possible to use a path with hash

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/compiler/src/aot/lazy_routes.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ function _collectLoadChildren(routes: string | Route | Route[], target: string[]
 
 export function parseLazyRoute(
     route: string, reflector: StaticReflector, module?: StaticSymbol): LazyRoute {
-  const [routePath, routeName] = route.split('#');
+  const [, routePath, routeName] = route.split(/(.*)#(.*)/);
```
Comment (IgorMinar) [inline packages/compiler/src/aot/lazy_routes.ts]: this means that a project name would be able to start with # but it won't able to contain # at any other position

Comment (gkalpak) [reply packages/compiler/src/aot/lazy_routes.ts]: What do you mean by _"project name"_?
The new RegExp will essentially treat everything before the last `#` as `routePath` and everything after the last `#` as `routeName` (but not sure if that is the desired behavior).

**Comment by petebacondarwin** [comment]
This PR looks like it is only testing ViewEngine (which is deprecated).
It would also need a test of the Ivy compiler before it can be reviewed.
Perhaps here: https://github.com/angular/angular/blob/3613e7c4e546f85dc8b0a88f66bcf619248fdd51/packages%2Fcompiler-cli%2Ftest%2Fngtsc%2Fngtsc_spec.ts#L5877

**Comment by mhevery** [comment]
Closing due to inactivity.

**Pull request closed without merging**